src/test/api/users.test.ts

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+import fetchApiData from '../../api/users';
+
+jest.mock('axios');
+
+describe('fetchApiData', () => {
+
+    // reset all mocked functions' states after each test
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    const responseData = [{ id: 1, title: 'Test Post' }];
+    const response = { data: responseData };
+
+    it('fetches successfully data from API', async () => {
+        (axios.get as jest.Mock).mockResolvedValue(response);
+
+        const data = await fetchApiData();
+        expect(data).toEqual(responseData);
+    });
+
+    it('handles API error', async () => {
+        const errorMeassage = 'Network Error';
+        (axios.get as jest.Mock).mockRejectedValue(new Error(errorMeassage));
+
+        try {
+            await fetchApiData();
+        } catch (error) {
+            if (error instanceof Error) {
+                expect(error.message).toEqual(errorMeassage);
+            }
```

Comment:
in this case I think if error is not Error type it should fail

Reply:
Yes Ram